### PR TITLE
Adding digital_ocean_space alias for s3_bucket module

### PR DESCRIPTION
**Pull request by slapula**
##### SUMMARY
This PR is based off of suggestions from #38774.  I am adding a symlink called `digital_ocean_space` that points to the `s3_bucket` module.  This means that using that alias will direct the AWS modules to use a default DigitalOcean endpoint if the user doesn't already specify it with the `s3_url` parameter.  This alias works for me in my basic testing but I could use some more eyes to see if I'm missing anything important.  One thing to note:  I do have some rudimentary integration testing for this alias and it involved filling in some stuff in the `ansible/test/runner` directory. I'm not sure if I should include that work here or if that would warrant another pull request.

@willthames @BondAnthony

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
`s3_bucket`
`digital_ocean_space`

##### ANSIBLE VERSION
```
ansible 2.4.1.0
  config file = None
  configured module search path = [u'/Users/asmith/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/asmith/.pyenv/versions/2.7.13/lib/python2.7/site-packages/ansible
  executable location = /Users/asmith/.pyenv/versions/2.7.13/bin/ansible
  python version = 2.7.13 (default, Sep  8 2017, 15:16:38) [GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)]
```

##### ADDITIONAL INFORMATION
N/A

### CODE REVIEW

File: lib/ansible/modules/cloud/amazon/s3_bucket.py
Comment (slapula) [inline lib/ansible/modules/cloud/amazon/s3_bucket.py]: This is... not ideal.  I did this to get around the fact that DigitalOcean doesn't seem to like adding the `LocationContraint` configuration parameter to CreateBucket requests.  I'm open to other ways of handling this.  This line will have to be updated if and when DO adds new regions with Spaces support.

**Comment by slapula** [comment]
I'm a little stumped on how to handle the doc-build errors.

File: lib/ansible/modules/cloud/amazon/s3_bucket.py
Comment (pmarques) [reply lib/ansible/modules/cloud/amazon/s3_bucket.py]: probably doing a condition to verify if **module._name** is **digital_ocean_space** will be better, otherwise, from your comment, if DO adds more regions this will fail.

**Review by pmarques — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/amazon/s3_bucket.py
Comment (slapula) [reply lib/ansible/modules/cloud/amazon/s3_bucket.py]: Yeah, that's the better solution.  I thought bringing in `module` would be an issue but its not... I was just overthinking it.  PR updated.

**Comment by mattclay** [comment]
I'm not sure if this aliasing behavior is something we want to support. I've added it to the agenda to be discussed in the next Ansible Project meeting.

**Comment by BondAnthony** [comment]
@mattclay I saw your notes from the meeting a couple weeks ago. I'll add this to the Ansible Project Meeting for Thursday.

**Comment by s-hertel** [comment]
Hi @slapula. I believe s3_bucket now supports DO since cfe7c9606c6fd628388aecc32127737d35c60877 was added. Can you verify that?

**Comment by kbreit** [comment]
@s-hertel @slapula I did a test a few days at ago and received an error. Please see the link below. I may try to use this branch and see if it works. 

https://github.com/ansible/ansible/pull/38774#issuecomment-442171219

**Comment by slapula** [comment]
Thanks @kbreit! What version are you testing with? Latest devel branch?

**Comment by kbreit** [comment]
@slapula Yes I was using devel as of a couple of days ago.

**Comment by gundalow** [comment]
Hi, just seeing what's needed to progress this (or https://github.com/ansible/ansible/pull/38774#issuecomment-442171219 )

**Comment by kbreit** [comment]
@gundalow Last I tried it worked. So I’m good with it as of when I tested. However, I didn’t develop the module so others input is welcome.

**Comment by BondAnthony** [comment]
We can merge this if the Ansible core team is okay with not having a single s3 module for the entire project. The last place we left off was testing walrus.

**Comment by gundalow** [comment]
Hi,
We've been chatting about this some more.

* We are OK with the examples being updated, showing how to use `s3_url`
* We are OK with updating `s3_url` documentation to detail how this could be used for `Ceph, DigitalOcean, Eucalypus, fakes3, etc`
* Adding aliases (symlink) isn't something we want to do. We feel that this could get into a mess over time. Also, it implicitly adds `support:core` to the DO version of this module, though that isn't something we can test/defend.


If @slapula would be OK to update this to be only a documentation update we can get it merged.

**Review by gundalow — CHANGES_REQUESTED** [review]
Please update to be a "docs only" update (no alias).

**Comment by slapula** [comment]
@gundalow Updated.  Let me know if there's anything else you'd like me to update.

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
Thank everybody, merged.

**Comment by BondAnthony** [comment]
Thank you all for getting these into the next release!!

**Comment by gundalow** [comment]
As this ended up being docs only update, I think we would accept a backport to `stable-2.7`, so feel free to create a backport PR.

**Pull request merged**
